src/LanguageServer/Protocol/LspServices/RequestTelemetryScope.cs

```diff
@@ -43,4 +47,15 @@ public override void Dispose()
 
         _telemetryLogger.UpdateTelemetryData(Name, Language, _queuedDuration, requestDuration, _result);
     }
+
+    private static void ReportNonFatalError(Exception exception)
+    {
+        if (exception is StreamJsonRpc.LocalRpcException localRpcException && localRpcException.ErrorCode == LspErrorCodes.ContentModified)
+        {
+            // Content modified exceptions are expected and should not be reported as NFWs.
```

Comment:
i'd love to know more about this.

Reply:
We throw `ContentModified` errors when the client asks us to resolve a codelens / inlay hint associated with a solution version we no longer have.   This can happen when project changes happen underneath us.  We eventually tell the client to refresh, but the client can send us resolve requests for the prior codelens/inlay hints before it sees the refresh.

We don't need to report NFW for these since they are expected to happen in normal workflows.